static/app/views/setupWizard/index.tsx

```diff
@@ -1,59 +1,332 @@
+    if (organizations.length === 1) {
+      return organizations[0].id;
+    }
+    // Pre-fill the last used org if there are multiple
+    if (lastOrganization) {
+      return lastOrganization.id;
+    }
+    return null;
+  });
+  const [selectedProjectId, setSelectedProjectId] = useState<string | null>(null);
+
+  const selectedOrg = useMemo(
+    () => organizations.find(org => org.id === selectedOrgId),
+    [organizations, selectedOrgId]
+  );
+
+  const orgProjectsRequest = useOrganizationProjects(selectedOrg);
+
+  const selectedProject = useMemo(
+    () => orgProjectsRequest.data?.find(org => org.id === selectedProjectId),
+    [orgProjectsRequest.data, selectedProjectId]
+  );
+
+  const {
+    mutate: updateCache,
+    isPending,
+    isSuccess,
+  } = useMutation({
+    mutationFn: (params: {organizationId: string; projectId: string}) => {
+      return baseApi.requestPromise(`/account/settings/wizard/${hash}/`, {
+        method: 'POST',
+        data: params,
+      });
+    },
+  });
+
+  const handleSubmit = useCallback(
+    (event: React.FormEvent) => {
+      event.preventDefault();
+      if (!selectedOrgId || !selectedProjectId) {
+        return;
+      }
+      updateCache(
+        {
+          organizationId: selectedOrgId,
+          projectId: selectedProjectId,
+        },
+        {
+          onError: () => {
+            addErrorMessage(t('Something went wrong! Please try again.'));
+          },
+        }
+      );
+    },
+    [selectedOrgId, selectedProjectId, updateCache]
+  );
+
+  const orgOptions = useMemo(
+    () =>
+      organizations
+        .map(org => ({
+          value: org.id,
+          label: org.name || org.slug,
+          leadingItems: <OrganizationAvatar size={16} organization={org} />,
+        }))
+        .toSorted((a, b) => a.label.localeCompare(b.label)),
     [organizations]
   );
-  const urlParams = new URLSearchParams(location.search);
-  const projectPlatform = urlParams.get('project_platform') ?? undefined;
 
-  const analyticsParams = useMemo(
-    () => ({
-      organization,
-      project_platform: projectPlatform,
-    }),
-    [organization, projectPlatform]
+  const projectOptions = useMemo(
+    () =>
+      (orgProjectsRequest.data || [])
+        .toSorted(
+          (a, b) => new Date(b.dateCreated).getTime() - new Date(a.dateCreated).getTime()
+        )
+        .map(project => ({
+          value: project.id,
+          label: project.name,
+          leadingItems: <ProjectBadge avatarSize={16} project={project} hideName />,
+        })),
+    [orgProjectsRequest.data]
+  );
+
+  const isFormValid = selectedOrg && selectedProject;
+
+  if (isSuccess) {
+    return <WaitingForWizardToConnect hash={hash} organizations={organizations} />;
+  }
+
+  return (
+    <StyledForm onSubmit={handleSubmit}>
+      <Heading>{t('Select your Sentry project')}</Heading>
+      <FieldWrapper>
+        <label>{t('Organization')}</label>
+        <StyledCompactSelect
```

Comment:
I spend an hour or so to write tests only to find out that as soon as I use the compact select, they fail with the message that a component update it not wrapped in `act()`.
I have not found the culprit, even if I render it without any of the props the error persists...